src/strings.c

```diff
@@ -2848,13 +2849,13 @@ skip_to_arg(
 
 	if (ap_types == NULL || ap_types[*arg_cur] == NULL)
 	{
-	    semsg(e_aptypes_is_null_str, fmt, *arg_cur);
+	    semsg(e_aptypes_is_null_str_nr, fmt, *arg_cur);
 	    return;
 	}
 
 	p = ap_types[*arg_cur];
 
-	int fmt_type = format_typeof(p, TRUE);
+	int fmt_type = format_typeof(p, usetvs);
```

Comment:
So, this should be as follows, I think.
```
	int fmt_type = format_typeof(p, FALSE);
```

Additionally, the following changes might be needed?
```diff
--- a/src/strings.c
+++ b/src/strings.c
@@ -2293,7 +2293,7 @@ format_typeof(
     {
 	switch (fmt_spec)
 	{
-	    case 'd': case 'u': case 'o': case 'x': case 'X':
+	    case 'd': case 'u': case 'o': case 'x': case 'X': case 'b': case 'B':
 		if (length_modifier == '\0')
 		    length_modifier = 'L';
 	}
```